docs/streaming.md

```diff
@@ -0,0 +1,510 @@
+# Streaming Code Generation
+
+The Qiskit Code Assistant supports **real-time streaming** for code completions, providing instant feedback as code is generated token-by-token.
+
+## Overview
+
+Streaming allows you to see code suggestions appear in real-time as they're being generated, similar to how modern AI chat interfaces work. This provides a more interactive and responsive experience compared to waiting for the entire completion.
+
+## Streaming Status
+
+Streaming is **enabled by default** starting with this version, providing the best user experience with real-time code generation.
```

Comment:
should the specific version be mentioned? otherwise, `starting with this version` may not be accurate or applicable in future versions

Reply:
I do agree. Documenting as if it would be released in v 0.14

Reply:
Done in https://github.com/Qiskit/qiskit-code-assistant-vscode/pull/151/commits/40db662bca563f331b12496574c5d47fc5f950db